aquarius/events/events_monitor.py

```diff
@@ -119,60 +118,83 @@ def __init__(self, web3, config_file):
             "Purgatory enabled" if self.purgatory else "Purgatory disabled"
         )
         logger.info(purgatory_message)
+        self._thread_process_blocks_is_on = False
+        self._thread_process_queue_is_on = False
+        self._thread_process_ve_allocate_is_on = False
+        self._thread_process_purgatory_is_on = False
 
     @property
     def block_envvar(self):
         return "METADATA_CONTRACT_BLOCK"
 
-    def start_events_monitor(self):
-        if self._monitor_is_on:
-            return
+    def stop_monitor(self):
+        """Stops all threads for processing more data"""
+        self._thread_process_blocks_is_on = False
+        self._thread_process_queue_is_on = False
+        self._thread_process_ve_allocate_is_on = False
+        self._thread_process_purgatory_is_on = False
 
-        logger.info("Starting the events monitor.")
-        t = Thread(target=self.run_monitor, daemon=True)
-        self._monitor_is_on = True
+    def start_events_monitor(self):
+        """Starts all needed threads, depending on config"""
+        logger.info("Starting the threads..")
+        t = Thread(target=self.thread_process_blocks, daemon=True)
+        self._thread_process_blocks_is_on = True
         t.start()
+        if strtobool(os.getenv("PROCESS_RETRY_QUEUE", "0")):
+            t = Thread(target=self.thread_process_queue, daemon=True)
+            self._thread_process_queue_is_on = True
+            t.start()
+        if self.ve_allocate:
+            t = Thread(target=self.thread_process_ve_allocate, daemon=True)
+            self._thread_process_ve_allocate_is_on = True
+            t.start()
+        if self.purgatory:
+            t = Thread(target=self.thread_process_purgatory, daemon=True)
+            self._thread_process_purgatory_is_on = True
+            t.start()
```

Comment:
self._monitor_is_on = True?

Reply:
no, I have removed _monitor_is_on and replaced it with 4 variables, one for each thread